pkg/controllers/mkstaskrun/controller.go

```diff
@@ -0,0 +1,69 @@
+package mkstaskrun
+
+import (
+	"fmt"
+
+	"github.com/MiniTeks/mks-server/pkg/apis/mkscontroller/v1alpha1"
+	clientset "github.com/MiniTeks/mks-server/pkg/client/clientset/versioned"
+	informer "github.com/MiniTeks/mks-server/pkg/client/informers/externalversions/mkscontroller/v1alpha1"
+	"github.com/MiniTeks/mks-server/pkg/tconfig"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+type Controller struct {
+	kubeclientset kubernetes.Interface
+	mksclientset  clientset.Interface
+}
+
+func NewController(kubeclientset kubernetes.Interface,
+	mksclientset clientset.Interface,
+	mksinformer informer.MksTaskRunInformer) *Controller {
+	controller := &Controller{
+		kubeclientset: kubeclientset,
+		mksclientset:  mksclientset,
+	}
+
+	mksinformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    addController,
+		UpdateFunc: updateController,
+		DeleteFunc: deleteController,
+	})
+
+	return controller
+}
+
+func addController(obj interface{}) {
+	fmt.Println("MksTaskRun has been created")
+
+	tp := &tconfig.TektonParam{}
```

Comment:
why you need these ?

Reply:
Tekton configuration wll be needed to communicate with Tekton API. I guess this is redundant and the configuration should be passed to NewController func instead of having one for each controller.